docs/specs/target.md

```diff
@@ -0,0 +1,182 @@
+architectures:
+
+- Targets MUST support the [agent architecture][agent_architecture] by
+  providing a single command that deploys Vector and achieves the
+  [agent architecture requirements](#agent-architecture).
+- Targets SHOULD support the [aggregator architecture][aggregator_architecture] by
+  providing a single command that deploys Vector and achieves the
+  [aggregator architecture requirements](#aggregator-architecture).
+- Targets MAY support the [unified architecture][unified_architecture] by
+  providing a single command that deploys Vector and achieves the
+  [unified architecture requirements](#unified-architecture).
+
+### 4. Agent Architecture
+
+The [agent architecture][agent_architecture] deploys Vector on each individual
+node for distributed data collection and processing. Along with general
+[hardening](#7-hardening) requirements, the following requirements define support
+for this architecture:
+
+- Architecture
+  - MUST deploy as a daemon on existing nodes, one Vector process per node.
+  - MUST deploy with Vector's [default agent configuration][default_agent_configuration].
+- Sizing
+  - MUST deploy as a good infrastructure citizen, giving resource priority to
+    other services on the same node.
+  - SHOULD be limited to 1 vCPUs by default, MUST be overridable by the user.
+  - SHOULD be limited to 2 GiB of memory per vCPU by default, MUST be
+    overridable by the user.
+  - SHOULD be limited to 1 GiB of disk space, MUST be overridable by the user.
+
+### 5. Aggregator Architecture
+
+The [aggregator architecture][aggregator_architecture] deploys Vector onto
+dedicated nodes for data aggregation. Along with general [hardening](#7-hardening)
+requirements, the following requirements define support for this architecture:
+
+- Architecture
+  - MUST deploy as a service with reserved/dedicated resources.
+  - SHOULD deploy with a persistent disk that is available between deployments by default,
+     MUST be overridable by the user if they do not want a persistent disk.
+  - MUST deploy with Vector's [default aggregator configuration][default_aggregator_configuration].
+  - Configured Vector ports, including non-default user configured ports,
+    SHOULD be automatically accessible within the Cluster or VPC.
+  - Configured Vector sources, including non-default user configured sources,
+    SHOULD be automatically discoverable via target service discovery
+    mechanisms.
+- Sizing
+  - MUST have dedicated/reserved resources that cannot be stolen by other services, preventing
+     the "noisy neighbor" problem to the degree possible.
+  - The Vector service SHOULD NOT be artificially limited with resource
+    limiters such as cgroups.
+  - SHOULD require 8 vCPUs by default, MUST be overridable by the user.
+  - SHOULD require 2 GiB of memory per vCPU (16 GiB in this case) by default,
+     MUST be overridable by the user.
+  - SHOULD request 36 GiB of disk space per vCPU by default (288 GiB in this case),
```

Comment:
Real glad for clarification on this one, I'll update my POV doc to reflect this.